fetch.bs

```diff
@@ -7440,27 +7439,100 @@ fetch("https://www.example.com/")
 
 <h3 id=websocket-connections>Connections</h3>
 
+<p>A user agent has an associated <dfn>connecting WebSocket connections</dfn>, a <a for=/>map</a> of
+(<a for=/>network partition key</a>, <a for=/>host</a>, <a for=url>port</a>) to a
+<a for=/>set</a> of unique identifiers.
+
+<p>A user agent has an associated <dfn>total pending proxy WebSocket connections</dfn>, a
+<a for=/>map</a> of <a for=/>network partition key</a> to an integer, starting at 0.
+
 <p>To <dfn id=concept-websocket-connection-obtain>obtain a WebSocket connection</dfn>, given a
-<var>url</var>, run these steps:
+<a>network partition key</a> <var>key</var>, <a for=/>origin</a> <var>origin</var>, and boolean
+<var>credentials</var>, run these steps:
 
 <ol>
- <li><p>Let <var ignore>host</var> be <var>url</var>'s <a for=url>host</a>.
+ <li><p>Let <var>proxies</var> be the result of performing proxy resolution for <var>url</var> in an
+ <a>implementation-defined</a> manner. If a proxy preference order has not been explicitly
+ configured then socks proxies should be preferred to https proxies which should be preferred to
+ http proxies.
+
+ <!-- Since this depends on the url it probably needs to be hoisted out of this algorithm. -->
```

Comment:
Why does this depend on the URL?

Reply:
The URL is passed to the `FindProxyForURL()` function in the proxy.pac file. proxy.pac can in principle choose a different proxy depending on the URL path, although apparently this [didn't work properly in IE](https://en.wikipedia.org/wiki/Proxy_auto-config#Old_Microsoft_problems).

Reply:
Ah, but that sounds like a problem for "obtain a connection" in general. I think we should add proxy support there instead and then layer this on top. Do you agree?

Reply:
Yes. I am planning to draft a change which moves most of the proxy parts to "obtain a connection". Unless you get to it first.